setup.py

```diff
@@ -42,6 +42,7 @@ def get_install_requires() -> List[str]:
         "sqlalchemy>=1.3.0",
         "tqdm",
         "PyYAML",  # Only used in `optuna/cli.py`.
+        'contextvars;python_version<"3.7"',
```

Comment:
Either one would be fine. I just chose `contextvars` since it is newly added to Python and is more sophisticated.
https://peps.python.org/pep-0567/#converting-code-that-uses-threading-local

Reply:
Definitely not a strong opinion but if the additional sophistication is not necessary, I'd prefer sticking to `threading.local` to keep dependencies simpler and possibly maintenance easier. The amount of code is basically the same I think. It might just be that I'm not very familiar with `contextvars`.